src/components/Search/SearchByNameResults.tsx

```diff
@@ -0,0 +1,48 @@
+import { useTranslation } from 'react-i18next'
+import { UDTQueryResult } from '../../services/ExplorerService/fetcher'
+import styles from './SearchByNameResults.module.scss'
+import EllipsisMiddle from '../EllipsisMiddle'
+
+type Props = {
+  udtQueryResults: UDTQueryResult[]
+  truncateTypeHash?: boolean
+}
+
+export const SearchByNameResults = (props: Props) => {
```

Comment:
This component becomes sluggish with a large amount of data.

Consider limiting the maximum number of rendered items or using a virtualized list.